test/parallel/test-timers-same-timeout-wrong-list-deleted.js

```diff
@@ -0,0 +1,54 @@
+'use strict';
+
+/*
+ * This is a regression test for https://github.com/nodejs/node/issues/7722.
+ *
+ * When nested timers have the same timeout, calling clearTimeout on the
+ * older timer after it has fired causes the list the newer timer is in
+ * to be deleted. Since the newer timer was not cleared, it still blocks
+ * the event loop completing for the duration of its timeout, however, since
+ * no reference exists to it in its list, it cannot be canceled and its
+ * callback is not called when the timeout elapses.
+ */
+
+require('../common');
+const assert = require('assert');
+const Timer = process.binding('timer_wrap').Timer;
+
+const TIMEOUT = 100;
+const start = Timer.now();
+
+function testNestedTimers() {
+  // This bug also prevents the erroneously dereferenced timer's callback
+  // from being called, so we can't use it's execution or lack thereof
+  // to assert that the bug is fixed.
+  process.on('exit', function() {
```

Comment:
Even if the `'exit'` event has a good chance of being emitted, and thus this callback will likely be called unless there's a serious bug caught by other tests, this callback would still be a good candidate to wrap in `common.mustCall()`.